.github/workflows/styler-actions.yml

```
@@ -0,0 +1,101 @@
+      - name: testing
+        run: echo '${{ steps.file_changes.outputs.files_modified}}'   
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/pr-fetch@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: r-lib/actions/setup-r@master
+      - name: Install dependencies
+        run: |
+          Rscript -e 'install.packages("styler")'
+          Rscript -e 'install.packages("devtools")'
+          Rscript -e 'devtools::install_version("roxygen2", version = "7.0.2", repos = "http://cran.us.r-project.org")'
+      - name: string operations
+        run: |
+          echo '${{ steps.file_changes.outputs.files_modified}}' > names.txt
+          cat names.txt | tr -d '[]' > new.txt
+          text=$(cat new.txt)
+          IFS=',' read -ra ids <<< "$text"
+          for i in "${ids[@]}"; do if [[ "$i" == *.R\" || "$i" == *.Rmd\" ]]; then echo "$i" >> new2.txt; fi; done
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v1
+        with:
+          name: artifacts
+          path: new2.txt
+      - name: Style
+        run: for i in $(cat new2.txt); do Rscript -e "styler::style_file("$i")"; done
+      - name: commit
+        run: |
+          git add \*.R
+          git add \*.Rmd
+          git commit -m 'Style'
+      - uses: r-lib/actions/pr-push@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          
+          
+  check:
+    needs: [style]
+    runs-on: ubuntu-latest
+    container: pecan/depends:develop
+    steps:
+      - name: check git version
+        id: gitversion
+        run: |
+          v=$(git --version | grep -oE '[0-9\.]+')
+          v='cat(numeric_version("'${v}'") < "2.18")'
+          echo "##[set-output name=isold;]$(Rscript -e "${v}")"
+      - name: upgrade git if needed
+        # Hack: actions/checkout wants git >= 2.18, rocker 3.5 images have 2.11
+        # Assuming debian stretch because newer images have git >= 2.20 already
+        if: steps.gitversion.outputs.isold == 'TRUE'
+        run: |
+          echo 'deb http://deb.debian.org/debian stretch-backports main' >> /etc/apt/sources.list
+          apt-get update
+          apt-get -t stretch-backports upgrade -y git
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/pr-fetch@master
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: r-lib/actions/setup-r@master
+      - name : download artifacts
+        uses: actions/download-artifact@v1
+        with:
+          name: artifacts
+      - name : make
+        shell: bash
+        run: |
+          cut -d / -f 1-2 artifacts/new2.txt | tr -d '"' > new.txt
+          cat new.txt
+          sort new.txt | uniq > uniq.txt
+          cat uniq.txt      
+          for i in $(cat uniq.txt); do make .doc/${i}; done
+      - name: commit
+        run: |
+          git config --global user.email "rahulagrawal799110@gmail.com"
+          git config --global user.name "Rahul Agrawal"
```

I don't know why but when I am using the container, git is complaining about the **config --global**. That's why I have added the following details.

---

The complaint is just because Git wants every commit to have someone's name associated with it, and the container starts out with no user configured. But you'll probably want to attribute these commits to a bot user rather than to yourself:

```suggestion
          git config --global user.email "pecan_bot@example.com"
          git config --global user.name "PEcAn stylebot"
```